registration-processor/core-processor/registration-processor-reprocessor-stage/src/main/java/io/mosip/registration/processor/reprocessor/stage/WorkflowSearchApi.java

```diff
@@ -0,0 +1,237 @@
+import io.mosip.registration.processor.core.workflow.dto.WorkFlowSearchResponseDto;
+import io.mosip.registration.processor.reprocessor.service.WorkflowSearchService;
+import io.mosip.registration.processor.rest.client.audit.builder.AuditLogRequestBuilder;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.RoutingContext;
+
+public class WorkflowSearchApi extends MosipVerticleAPIManager {
+
+
+	@Value("${vertx.cluster.configuration}")
+	private String clusterManagerUrl;
+
+	/** server port number. */
+	@Value("${mosip.regproc.workflowsearch.server.port}")
+	private String port;
+
+	/** worker pool size. */
+	@Value("${worker.pool.size}")
+	private Integer workerPoolSize;
+
+	private static final String STAGE_PROPERTY_PREFIX = "mosip.regproc.reprocessor.";
+
+	@Value("${mosip.regproc.workflowsearch.eventbus.port}")
+	private String eventBusPort;
+
+	@Value("${mosip.registration.processor.datetime.pattern}")
+	private String dateTimePattern;
+
+	@Value("${mosip.regproc.workflow.search.api-id}")
+	private String id;
+
+	@Value("${mosip.regproc.workflow.search.version}")
+	private String version;
+
+	@Autowired
+	MosipRouter router;
+
+	/**
+	 * The context path.
+	 */
+	@Value("${server.servlet.path}")
+	private String contextPath;
+
+	private MosipEventBus mosipEventBus = null;
+
+
+	/** The reg proc logger. */
+	private static Logger regProcLogger = RegProcessorLogger.getLogger(WorkflowSearchApi.class);
+
+	/** The module id. */
+	public static String MODULE_ID = PlatformSuccessMessages.RPR_WORKFLOW_ACTION_API_SUCCESS.getCode();
+
+	/** The module name. */
+	public static String MODULE_NAME = ModuleName.WORKFLOW_ACTION_API.toString();
+
+	/** The core audit request builder. */
+	@Autowired
+	AuditLogRequestBuilder auditLogRequestBuilder;
+
+	@Autowired
+	WorkflowSearchService workflowSearchService;
+
+	/**
+	 * Deploy verticle.
+	 */
+	public void deployVerticle() {
+		mosipEventBus = this.getEventBus(this, clusterManagerUrl, workerPoolSize);
+
+
+	}
+
+	@Override
+	public void start() {
+		router.setRoute(this.postUrl(getVertx(), null, null));
+		this.routes(router);
+		this.createServer(router.getRouter(), Integer.parseInt(port));
+	}
+
+	@Override
+	public Integer getEventBusPort() {
+		return Integer.parseInt(eventBusPort);
+	}
+
+	/**
+	 * contains all the routes in this stage
+	 *
+	 * @param router
+	 */
+	private void routes(MosipRouter router) {
+		router.post(contextPath + "/workflow/search");
+		router.handler(this::processURL, this::failure);
+	}
+	/**
+	 * method to process the context received.
+	 *
+	 * @param ctx the ctx
+	 */
+	public void processURL(RoutingContext ctx) {
+
+		String workflowSearch = null;
+		boolean isTransactionSuccessful = false;
+		LogDescription description = new LogDescription();
+		try {
+		JsonObject obj = ctx.getBodyAsJson();
+		String user = getUser(ctx);
+		SearchRequestDto searchRequestDTO = (SearchRequestDto) JsonUtils.jsonStringToJavaObject(SearchRequestDto.class,

```

Comment:
Please perform all the request object validations here, like the pagination info is available etc..

Reply:
taken care